core/src/main/scala/org/apache/spark/scheduler/Stage.scala

```diff
@@ -116,4 +116,7 @@ private[scheduler] abstract class Stage(
 
   /** Returns the sequence of partition ids that are missing (i.e. needs to be computed). */
   def findMissingPartitions(): Seq[Int]
+
+  def isIndeterminate: Boolean =
+    rdd.outputDeterministicLevel == DeterministicLevel.INDETERMINATE
```

Comment:
nit: if it needs to cross the line, it's better to write
```
def f(): T = {
  ...
}
```

Reply:
Thanks, done in 5a4b8b8.